packages/token-bridge-sdk/src/hooks/useBalance.spec.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { act, renderHook } from '@testing-library/react-hooks'
+import { useBalance } from './useBalance'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { BigNumber } from 'ethers'
+import { SWRConfig } from 'swr'
+import { PropsWithChildren } from 'react'
+
+// Create a new cache for every test
+const Container = ({ children }: PropsWithChildren<{}>) => (
+  <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
+)
+
+describe('useBalance', () => {
+  afterEach(() => {
+    jest.restoreAllMocks()
+  })
+
+  it('getter return null for undefined walletAddress', async () => {
+    const provider = new StaticJsonRpcProvider(
+      process.env.REACT_APP_ETHEREUM_RPC_URL,
+      1
+    )
+
```

Comment:
Should we also mock `provider.getBalance` here to make sure it's not a false positive?

Reply:
Make sense, I mocked `provider.getBalance` in the first two tests to be safe